_posts/2018-08-30-pair-programming.markdown

```diff
@@ -0,0 +1,184 @@
+of extra commits, or even re-wrote the entire pull request. That's when pairing becomes very valuable to your team.
+
+Another good example is on-boarding new hires. Coming from an agile consultancy and a photo-product company, I was
+nowhere close to an Art expert when I joined Artsy. This was the time I wanted to question not just technical
+questions, but also very basic questions about art. How do people find and purchase artworks? Who actually sells
+artworks? What is an art fair? And auctions? Christina Thompson, on the other hand, has been at Artsy for more than
+two years. She also has strong experience in agile software development and practices, including pair-programming.
+Naturally, we started pairing to familiarize myself both with the code base and basic knowledge about Art on my first
+week at Artsy.
+
+### When not to pair program?
+
+This post is about pair-programming, so why even talk about when not to pair-program? Well, while it’s a powerful tool,
+it's not a silver bullet we can utilize to solve all software problems. As we all deal with complex, long-standing
+pull requests, there are also simple changes that don't require a lot of discussions. Occasionally, your pull request
+is very long with a full of deprecated method names. Sometimes, you and your pair both feel stuck because none of you
+have context or knowledge about what you are addressing. Whenever you are not feeling as productive as you think you
+should be, then you don’t have to pair. Sometimes it may make sense for two of you to do research individually and
+check in later to share findings and learnings. In an occasional case, even swapping a pair is also a good way to make
+the entire team more productive.
+
+<!---------------------------------------------- I wonder if this "I believe" 👇 is okay -->
+
+So what is the key point that makes a pair-programming session successful? I believe pair-programming works best when
+there is a fair amount of knowledge gap between two people. One is coding and teaching simultaneously, and the
+other is reviewing and learning simultaneously. Here the crucial part is _teaching_. It is easy, especially for more
+seasoned developers, to ignore the opportunity to share thoughts and knowledge with new hires, losing productivity
+they could've gained by pairing. Joining a new company is always scary, and pair-programming with a new hire will
+avoid a lot of "I wasn't sure about X" moments new hires might have encountered. At the same time, as mentioned above,
+pair-programming is not a silver bullet. It is important to be able to use pair-programming as a tool in your toolbox
+to solve a particular issue in your engineering team.
+
+## What You'll Need to Pair Program
+
+The minimum you need is a computer and the ability to communicate, but let's look at what an optimal pair
+programming setup looks like.
+
+### Offline
+
+If you can have physical access to the other engineer, then you should aim to have:
```

Comment:
> If you can have physical access to the other engineer

This makes it sound like the other engineer is a computer 😂 Maybe "if you and the other engineer are in the same physical space"?